src/tables.jl

```diff
@@ -165,6 +171,14 @@ function read(io_or_path; validate::Bool=true)
                                              via `Legolas.read`; is it missing the expected custom metadata and/or the
                                              expected \"$LEGOLAS_SCHEMA_QUALIFIED_METADATA_KEY\" field?
                                              """))
+
+        provider = extract_metadata(table, LEGOLAS_SCHEMA_PROVIDER_METADATA_KEY)
+        # If we don't have the schema defined in our session (i.e. `Legolas.schema_provider` is `nothing`), 
+        # but we do have a hint of where the schema was defined via the metadata, then throw an informative
+        # error. If we don't error now, we will throw an `UnknownSchemaVersionError` with less information later.
+        if Legolas.schema_provider(Val(Legolas.name(sv))) === nothing && provider !== nothing
```

Comment:
why can't we just dispatch on the schema version itself?  seems weird to introduce `Val`-based dispatch here when we have the sv already from the parsed identifier in the metadata..

Reply:
I did it in the `@schema` macro rather than the `@version` macro so it's independent of version, but I could move it to `@version` and use the full sv